### PR TITLE
Branch budget checker

### LINKED ISSUE #70: Budget Checker for Pairing command
Checks if the client's budget is within the property's rental price before pairing them. If budget exceeded, no pairing occurs

### PR DESCRIPTION
**Pull request by ngdeqi**
Fixes #70

### CODE REVIEW

File: src/test/java/seedu/duke/PairingListTest.java [hunk 1]
```diff
@@ -100,6 +108,13 @@ void isAlreadyPaired_presentAndAbsentPropertyAndClients_success() {
         }
     }
 
+    @Test void hasPriceExceededBudget_differentBudgetClients_success() {
+        PairingList pairingList = pairingListInit();
+        assertTrue(pairingList.hasPriceExceededBudget(BUDGET_CHECKER_CLIENT_POOR, PRESENT_PROPERTY));
+        assertFalse(pairingList.hasPriceExceededBudget(PRESENT_CLIENT, PRESENT_PROPERTY));
+        assertFalse(pairingList.hasPriceExceededBudget(BUDGET_CHECKER_CLIENT_RICH, PRESENT_PROPERTY));
```
Comment (wilsonngja) [inline src/test/java/seedu/duke/PairingListTest.java:115]: I like the extensive test done to this feature.

File: src/main/java/seedu/duke/CommandStructure.java [hunk 2]
```diff
@@ -10,6 +10,8 @@ public class CommandStructure {
     public static final String COMMAND_CHECK = "check";
     public static final String COMMAND_EXIT  = "quit";
 
+    public static final String PROPERTY_FLAG = "-property";
```
Comment (wilsonngja) [inline src/main/java/seedu/duke/CommandStructure.java:13]: Good use of constant to represent the property and client flag.

File: src/main/java/seedu/duke/Messages.java [hunk 3]
```diff
@@ -113,10 +113,14 @@ public class Messages {
     public static final String MESSAGE_EXISTING_PAIR = "OOPS!! This client and this property are already paired "
             + "together. You don't need to pair them again.";
 
+    public static final String MESSAGE_BUDGET_EXCEEDED = "OOPS!! The rental price exceeds the client's budget. "
```
Comment (wilsonngja) [inline src/main/java/seedu/duke/Messages.java:116]: Good job in arranging the code in the right class

File: src/main/java/seedu/duke/PairingList.java [hunk 4]
```diff
@@ -120,6 +120,21 @@ public boolean isAlreadyPaired(Client client, Property property) {
...
 
+    /**
+     * Returns true if the rental price of the property exceeds the client's budget.
+     *
+     * @param client Client to be paired with the property.
+     * @param property Property to be paired with the client.
+     * @return True is rental price exceeds client's budget; false if rental price is equal to or lower than clien's
+     *         budget.
+     */
+
```
Comment (wilsonngja) [inline src/main/java/seedu/duke/PairingList.java:131]: Take note of the space here. After the Javadocs, there should not be an empty space.

**Review by wilsonngja — COMMENTED** [review]
Good work in implementing this function and following the coding standard. But do take note of the minor coding standard violation 👍🏼

**Review by OVReader — APPROVED** [review]
LGTM

**Pull request merged**